data/src/main/java/com/linkedin/data/schema/resolver/FileDataSchemaResolver.java

```diff
@@ -241,7 +265,7 @@ protected DataSchemaLocation transform(String path)
           StringBuilder builder = new StringBuilder();
           // within a JAR file, files are treated as resources. Thus, we should lookup using the resource separator
           // character, which is '/'
-          builder.append(InternalConstants.PEGASUS_DIR_IN_JAR)
+          builder.append(_schemasPathSuffix)
```

Comment:
We support /pegasus directory by default. Now we also want to support /extensions directory if configuration is provided. What will happen if user pass neither "pegasus" nor "extensions" in the set method? Shall we provide some check here? Also we should update this information in the set method java doc, so users could know what directories are supported.

Reply:
_schemasPathSuffix is initialized with default "pegasus". Had added the comments on field now added the same on setter method

Reply:
The default one can be overridden by any value passed in the set method. For example, resolver.setSchemaDirectory("") or resolver.setSchemaDirectory("foo"). We don't provide any constrains, just pass whatever user pass in the set method and use it as the directory doesn't seem right. Any schemas placed at root or alternative root directory other than "pegasus" and "extensions" should be ignored. 
Maybe we should introduce an enum, use that one as the setter value.

Reply:
I think from the security point of view, we can gather allowable strings in allowed enums, so that other content won't be leaked.